src/TheThingsNetwork.cpp

```diff
@@ -312,3 +312,55 @@ void TheThingsNetwork::showStatus() {
   debugPrint(F("RX Delay 2: "));
   debugPrintLn(readValue(F("mac get rxdelay2")));
 }
+
+void TheThingsNetwork::configure_channels(int total_ch) {
+   int ch;
+   long int freq = this->fp;
+   String str = "";
+
+   for (ch = 0; ch < total_ch; ch++) {
+       str.concat(F("mac set ch freq "));
+       str.concat(ch);
+       str.concat(F(" "));
+       str.concat(freq);
+       sendCommand(str);
+       str = "";
+       str.concat(F("mac set ch drrange "));
+       str.concat(ch);
+       str.concat(F(" 0 5"));
+       sendCommand(str);
+       str = "";
+       str.concat(F("mac set ch status "));
+       str.concat(ch);
+       str.concat(F(" on"));
+       sendCommand(str);
+       str = "";
+       freq = freq + 200000;
+   }
+}
+
+void TheThingsNetwork::set_dcycle() {
```

Comment:
Switch/case for `fp` and ask @tftelkamp for settings per frequency plan. Start with EU868 and US915